go/arrow/datatype_nested.go

```diff
@@ -288,7 +288,12 @@ func (t *StructType) String() string {
 	return o.String()
 }
 
-func (t *StructType) Fields() []Field   { return t.fields }
+func (t *StructType) Fields() []Field {
```

Comment:
Can you add a godoc comment on the methods to indicate to callers that this will make a copy of the Fields?

Reply:
1e81c723eb665da8f59519710c749e0fda435676